### DISCUSSION TITLE
conditional removal

### DISCUSSION THREAD

**Discussion by twobirdshealing** [impress-org/givewp#6547] [Developer Q&A]
Iv been working with priority support and they seem to be stuck so Alex Coleman from GiveWP asked me to post here "Your best bet would be to submit this to the GitHub discussions page here: https://github.com/impress-org/givewp/discussions. Our developer see that and will be able to get a better answer for you." This is the code we are trying to get to work. It should do the css removal on only forms in the category "retreat" but is doing it on ALL forms no matter the category. Support is stumped as they helped provide the code and have tried figuring out why its not workin but no go. 

```php
<?php

function override_iframe_template_styles_with_inline_styles_cat()
{
    $objects = get_terms('give_forms_category');
    $cat_slug = 'slug';
    $value = 'retreat';
    $in_category = in_array($value, array_column($objects, $cat_slug));

    if ($in_category) {
        wp_add_inline_style(

            'give-classic-template',
            '
.give-form-stats-panel-list li:first-of-type {
display: none;
}
'
        );
    }
}

add_action('wp_print_styles', 'override_iframe_template_styles_with_inline_styles_cat', 10);

```

**Comment by kjohnson** [comment]
@twobirdshealing From what I can tell the code functions as intended.

I would suggest inspecting the `WP_Terms` returned by `get_terms()` to check that the values are what you expect.

```diff
<?php

function override_iframe_template_styles_with_inline_styles_cat()
{
    $objects = get_terms('give_forms_category');

+    echo '<pre>';
+    var_dump($objects);
+    echo '</pre>';
+    exit;

    // ...
}
```

**Comment by twobirdshealing** [comment]
Thanks I should have mentioned we tried that as well with support.? For example here are two separate forms one with a category set for "retreat" and one with category set as "donation" if you look at the pages I have the var dump turned on they are identical almost. This is where support got stuck and suggested I check here incase it was a bug issue. 

https://staging.twobirdschurch.com/donations/form-test-category-no-retreat/
https://staging.twobirdschurch.com/donations/form-test-category-retreat/

**Comment by kjohnson** [comment]
@twobirdshealing you may need to use `get_the_terms()` to return the terms for a particular post, see https://developer.wordpress.org/reference/functions/get_the_terms/. That may be what is missing here.
